src/js/actions/transform.js

```diff
@@ -1084,22 +1084,45 @@ define(function (require, exports) {
                 });
         }, this);
 
-        _layerTransformHandler = synchronization.debounce(function (event) {
+        _layerTransformHandler = function (event) {
             this.dispatch(events.ui.TOGGLE_OVERLAYS, { enabled: true });
-            
+
             var appStore = this.flux.store("application"),
-                currentDoc = appStore.getCurrentDocument(),
-                artboardsSelected = currentDoc.layers.selected.some(function (layer) {
-                    return layer.isArtboard;
-                });
-                
-            // If it was a simple click/didn't move anything, there is no need to update bounds
-            // But, if we're moving multiple artboards, we get one event with 0,0, which means
-            // we hit this and we have to update bounds
-            if (event.trackerEndedWithoutBreakingHysteresis && !artboardsSelected) {
-                return Promise.resolve();
-            }
+                currentDoc = appStore.getCurrentDocument();
 
+            // Handle the normal move events with a debounced function
+            var debouncedMoveHandler = synchronization.debounce(function () {
+                // short circuit based on this trackerEndedWithoutBreakingHysteresis event flag
+                if (event.trackerEndedWithoutBreakingHysteresis) {
```

Comment:
Actually, can you go even further and leave a comment that describes what events you anticipate, in what order, and what they all mean?